lib/passport-wsfed-saml2/utils.js

```diff
@@ -62,12 +63,8 @@ exports.getReqUrl = function(req){
 };
 
 exports.generateUniqueID = function() {
-  var chars = "abcdef0123456789";
-  var uniqueID = "";
-  for (var i = 0; i < 20; i++) {
-    uniqueID += chars.substr(Math.floor((Math.random()*15)), 1);
-  }
-  return uniqueID;
+  var uniqueID = crypto.randomBytes(10);
```

Comment:
Ah I didn't know that, thanks! I used 10 to end up with an id of the same length as before (20 digits) because I didn't know if that's some sort of requirement. Do you think changing the length won't break things?

Reply:
No, the ID is only used for preventing replay attacks and associating SAMLResponse to the SAMLRequest. As long as the ID is xml safe (which hex is), there should be no issues.

Reply:
Thanks for the info 👍 I'm updating.